modules/javafx.graphics/src/test/java/test/com/sun/javafx/iio/ImageStorageTest.java

```diff
@@ -57,27 +59,40 @@ public void createImageFromNoExtensionURL() throws ImageStorageException {
         assertNotNull(new ImageStorage().loadAll(path, null, 0, 0, true, 2.0f, true));
     }
 
-    @Test
-    public void testImageNames() {
+    @ParameterizedTest
+    @ValueSource(ints = {2, 3, 4})
+    public void testImageNames(int scale) {
         String [][]imageNames = new String[][] {
-            { "image", "image@2x" },
-            { "image.ext", "image@2x.ext" },
-            { "dir/image", "dir/image@2x" },
-            { "/dir.ext/image.ext", "/dir.ext/image@2x.ext" },
-            { "file:image", "file:image@2x" },
-            { "file:image.ext", "file:image@2x.ext" },
-            { "http://test.com/image", "http://test.com/image@2x" },
-            { "http://test.com/dir.ext/image", "http://test.com/dir.ext/image@2x" },
-            { "http://test.com/image.ext", "http://test.com/image@2x.ext" },
-            { "http://test.com/dir.ext/image.ext", "http://test.com/dir.ext/image@2x.ext" },
+            { "image", "image@" + scale + "x" },
+            { "image.ext", "image@" + scale + "x.ext" },
+            { "dir/image", "dir/image@" + scale + "x" },
+            { "/dir.ext/image.ext", "/dir.ext/image@" + scale + "x.ext" },
+            { "file:image", "file:image@" + scale + "x" },
+            { "file:image.ext", "file:image@" + scale + "x.ext" },
+            { "http://test.com/image", "http://test.com/image@" + scale + "x" },
+            { "http://test.com/dir.ext/image", "http://test.com/dir.ext/image@" + scale + "x" },
+            { "http://test.com/image.ext", "http://test.com/image@" + scale + "x.ext" },
+            { "http://test.com/dir.ext/image.ext", "http://test.com/dir.ext/image@" + scale + "x.ext" },
         };
         for (String[] names : imageNames) {
-            String name2x = ImageTools.getScaledImageName(names[0]);
+            String name2x = ImageTools.getScaledImageName(names[0], scale);
```

Comment:
Minor: maybe rename this variable to `nameScaled` or similar (since it isn't always 2x any more)?